File: cogs/ban_member.py
from discord.ext import commands
from discord import app_commands
import discord


class ban_member(commands.Cog):
    def __init__(self, bot: commands.Bot) -> None:
        self.bot: commands.Bot = bot

    @app_commands.command(name="ban_member", description="Banされたユーザー一覧を表示します。")
    async def ban_members(self, i: discord.Interaction):
        m = []
        async for entry in i.guild.bans(limit=999999):
            m.append(f"{entry.user.name}, ")
        if not len(m) == 0:
            end = len(m) - 1
            m[end] = m[end].split(", ")[0]
            send_content = "".join(m)
        elif len(m) == 0:
            send_content = "Banされたユーザーはいません。"
        try:
            await i.response.send_message(
                embed=discord.Embed(
                    title="Banされたユーザー", description=send_content, color=0x3498DB
                ),
                ephemeral=True,
            )
        except:
            await i.response.send_message(
                embed=discord.Embed(
                    title="Banされたユーザー", description="取得できませんでした。", color=0x3498DB
                ),
                ephemeral=True,
            )


async def setup(bot: commands.Bot) -> None:
    await bot.add_cog(ban_member(bot))

File: src/command/banmember.ts
import { Embed, Interaction } from "../deps/deps.ts";

export class BanMember {
  constructor() {}
  async run(interaction: Interaction) {
    if (!interaction.isApplicationCommand()) return;
    if (interaction.guild) {
      const bans = await interaction.guild.bans.all();
      const str =
        bans.map((ban) => ban.user.tag).join(", ") ||
        "Banされたユーザーがいません。";
      if (str.length > 4096) {
        const str2 = str.substring(4096, str.length);
        await interaction.reply({
          embeds: [
            new Embed()
              .setTitle("Banされたユーザー一覧")
              .setDescription(str)
              .setColor(0x3498db),
            new Embed()
              .setTitle("Banされたユーザー一覧")
              .setDescription(str2)
              .setColor(0x3498db),
          ],
        });
      } else {
        await interaction.reply({
          embeds: [
            new Embed()
              .setTitle("Banされたユーザー一覧")
              .setDescription(str)
              .setColor(0x3498db),
          ],
        });
      }
    } else {
      await interaction.reply({
        content: "ここはサーバーではありません。",
        ephemeral: true,
      });
    }
  }
}
